lib/gui/modules/image-writer.js

```diff
@@ -48,14 +76,134 @@ const analytics = require('../modules/analytics')
-      validateWriteOnSuccess: settings.get('validateWriteOnSuccess'),
-      unmountOnSuccess: settings.get('unmountOnSuccess')
+    ipc.server.on('error', (error) => {
+      terminateServer()
+      const errorObject = _.isError(error) ? error : errors.fromJSON(error)
+      reject(errorObject)
+    })
+
+    ipc.server.on('log', (message) => {
+      console.log(message)
+    })
+
+    const flashResults = {}
+    ipc.server.on('done', (results) => {
+      _.merge(flashResults, results)
+    })
+
+    ipc.server.on('state', onProgress)
+
+    const argv = _.attempt(() => {
+      const entryPoint = getApplicationEntryPoint()
+
+      // AppImages run over FUSE, so the files inside the mount point
+      // can only be accessed by the user that mounted the AppImage.
+      // This means we can't re-spawn Etcher as root from the same
+      // mount-point, and as a workaround, we re-mount the original
+      // AppImage as root.
+      if (os.platform() === 'linux' && process.env.APPIMAGE && process.env.APPDIR) {
+        return [
+          process.env.APPIMAGE,
+
+          // Executing the AppImage with ELECTRON_RUN_AS_NODE opens
+          // the Node.js REPL without loading the default entry point.
+          // As a workaround, we can pass the path to the file we want
+          // to load, relative to the usr/ directory of the mounted
+          // AppImage.
+          _.replace(entryPoint, path.join(process.env.APPDIR, 'usr/'), '')
+        ]
+      }
+
+      return [
+        _.first(process.argv),
+        entryPoint
+      ]
     })
-    child.on('error', reject)
-    child.on('done', resolve)
-    child.on('progress', onProgress)
+
+    ipc.server.on('start', () => {
+      console.log(`Elevating command: ${_.join(argv, ' ')}`)
+
+      permissions.elevateCommand(argv, {
+        applicationName: packageJSON.displayName,
+        environment: {
+          IPC_SERVER_ID,
+          IPC_CLIENT_ID,
+          IPC_SOCKET_ROOT: ipc.config.socketRoot,
+          ELECTRON_RUN_AS_NODE: 1,
+
+          // Casting to Number nicely converts booleans to 0 or 1.
+          OPTION_VALIDATE: Number(settings.get('validateWriteOnSuccess')),
+          OPTION_UNMOUNT: Number(settings.get('unmountOnSuccess')),
+
+          OPTION_IMAGE: image,
+          OPTION_DEVICE: drive.device,
+
+          // This environment variable prevents the AppImages
+          // desktop integration script from presenting the
+          // "installation" dialog
+          SKIP: 1
+        }
+      }).then((results) => {
+        flashResults.cancelled = results.cancelled
+        console.log('Flash results', flashResults)
+
+        // This likely means the child died halfway through
+        if (!flashResults.cancelled && !flashResults.bytesWritten) {
+          throw errors.createUserError({
+            title: 'The writer process ended unexpectedly',
+            description: 'Please try again, and contact the Etcher team if the problem persists',
```

Comment:
Sorry if this is a stupid question, but why does the identical string appear both here and in `lib/shared/messages.js` ?